### PR TITLE
[chore] add license-check

### PR DESCRIPTION
**Pull request by codeboten**
First pass at cleaning up the license in files.

### CODE REVIEW

**Comment by dmitryax** [comment]
We already have `addlicense` and `checklicense` targets in Makefile.common. Do we really need another one?

I think the problem with the existing `checklicense` is that it checks only for _presence_ of the license but doesn't compare the header line with what we want. I think we need to remove `-check` from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/Makefile.Common#L131 and use the diff approach that we have for other generated stuff. WDYT?

**Comment by codeboten** [comment]
> We already have `addlicense` and `checklicense` targets in Makefile.common. Do we really need another one?
> 

We likely don't need another one. I just used the check license from the opentelemetry-go-build-tools repo. I don't care which way we end up going, but i do like the simplicity of the check w/ awk

**Comment by codeboten** [comment]
After testing this out, i don't think addlicense will tell us if a license is different than what's expected. I ran it on the repo as is, and although it added licenses to any missing files, it left these in place (note the 2 spaces after Copyright):

```
// Copyright  The OpenTelemetry Authors
```

It's possible I misused `addlicence` though

File: .github/workflows/scripts/dependabot-pr.sh [hunk 1]
```diff
@@ -1,4 +1,17 @@
 #!/bin/bash -ex
+# Copyright The OpenTelemetry Authors
```
Comment (dmitryax) [inline .github/workflows/scripts/dependabot-pr.sh:2]: We have a line break here in contrib scripts. Should we do the same in core?

Comment (codeboten) [reply .github/workflows/scripts/dependabot-pr.sh]: Done

File: .github/workflows/scripts/dependabot-pr.sh [hunk 2]
```diff
@@ -1,4 +1,18 @@
 #!/bin/bash -ex
+#
+# Copyright The OpenTelemetry Authors
```
Comment (bogdandrutu) [inline .github/workflows/scripts/dependabot-pr.sh:3]: Should this be before the first line?

**Review by bogdandrutu — COMMENTED** [review]
LGTM on the fixes, I will let you folks @dmitryax and @codeboten to decide which tool to use.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**